api/api.go

```diff
@@ -59,30 +59,18 @@ func Setup(ctx context.Context, cfg *config.Config, router *mux.Router, dataStor
 
 		log.Info(ctx, "enabling only public endpoints for dataset api")
 		api.enablePublicEndpoints(ctx)
-
-		if cfg.EnablePermissionsAuth {
```

Comment:
Also, I would suggest to do 

`if api.enablePrivateEndpoints` instead of `if cfg.EnablePrivateEndpoints` so that we are using the `api` object completely when setting up for the API

Reply:
If im doing this then i will remove all this logic as explained in the above - the app is way more complicated then it needs to be right now